lib/matplotlib/axes/_axes.py

```diff
@@ -2738,20 +2736,12 @@ def stem(self, *args, linefmt=None, markerfmt=None, basefmt=None, bottom=0,
 
         # New behaviour in 3.1 is to use a LineCollection for the stemlines
         if use_line_collection:
-            if orientation == 'horizontal':
-                stemlines = [
-                    ((bottom, loc), (head, loc))
-                    for loc, head in zip(locs, heads)]
-            else:
-                stemlines = [
-                    ((loc, bottom), (loc, head))
-                    for loc, head in zip(locs, heads)]
             if linestyle is None:
                 linestyle = rcParams['lines.linestyle']
-            stemlines = mcoll.LineCollection(stemlines, linestyles=linestyle,
-                                             colors=linecolor,
-                                             label='_nolegend_')
-            self.add_collection(stemlines)
+            xlines = self.vlines if orientation == "vertical" else self.hlines
+            stemlines = xlines(
```

Comment:
Just to be sure:

~~~
            self.update_datalim(corners)
            self._request_autoscale_view()
~~~
used in xlines does not have any unintended side-effect?

Reply:
We are always going to plot the markers and the baseline, which necessarily bracket the stemlines, so the autolims set by the markers and baseline will be as big as those set by the stemlines anyways.